p2p/pex/addrbook.go

```diff
@@ -294,29 +295,32 @@ func (a *addrBook) MarkBad(addr *p2p.NetAddress) {
 
 // GetSelection implements AddrBook.
 // It randomly selects some addresses (old & new). Suitable for peer-exchange protocols.
+// Must never return a nil address.
 func (a *addrBook) GetSelection() []*p2p.NetAddress {
 	a.mtx.Lock()
 	defer a.mtx.Unlock()
 
-	if a.size() == 0 {
+	bookSize := a.size()
+	if bookSize == 0 {
```

Comment:
I get quite nervous when I see `== 0` for checks that result from additions of two entities that are independently subtracted, are ints and moreover don't have full guarantees to always be a minimum of 0.

Perhaps let's make that check `<= 0`?

I personally only trust the Go guarantee of `len` returning a minimum of 0 always since the compiler, runtime and spec guarantee that, but the rest nah.